From: Juergen Borleis <jbe@pengutronix.de>
Date: Tue, 1 Dec 2015 12:33:04 +0100
Subject: [PATCH] DTS/network: Network: setup pins for a high speed connection

100 MHz aware pins and some FEC related GPIO are required.

Signed-off-by: Juergen Borleis <jbe@pengutronix.de>
---
 arch/arm/boot/dts/imx6qdl-cpu.dtsi | 25 ++++++++++++++-----------
 1 file changed, 14 insertions(+), 11 deletions(-)

diff --git a/arch/arm/boot/dts/imx6qdl-cpu.dtsi b/arch/arm/boot/dts/imx6qdl-cpu.dtsi
index 823db91eb65b..464e4a68eb01 100644
--- a/arch/arm/boot/dts/imx6qdl-cpu.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-cpu.dtsi
@@ -340,9 +340,7 @@
 	pinctrl_board: dol63x_grp {
 		pinctrl_hog: hoggrp {
 			fsl,pins = <
-				MX6QDL_PAD_GPIO_5__GPIO1_IO05		0xb0		/* ethernet phy reset */
 				MX6QDL_PAD_EIM_D30__GPIO3_IO30		0xb0		/* ethernet phy interrupt */
-				MX6QDL_PAD_ENET_RX_ER__GPIO1_IO24	0x13000		/* Enet */
 			>;
 		};
 
@@ -395,15 +393,20 @@
 
 		pinctrl_enet_4: enetgrp-4 {
 			fsl,pins = <
-				MX6QDL_PAD_ENET_MDIO__ENET_MDIO		0x100b0
-				MX6QDL_PAD_ENET_MDC__ENET_MDC		0x100b0
-				MX6QDL_PAD_ENET_CRS_DV__ENET_RX_EN	0x100b0
-				MX6QDL_PAD_ENET_TX_EN__ENET_TX_EN	0x100b0
-				MX6QDL_PAD_ENET_RXD0__ENET_RX_DATA0	0x100b0
-				MX6QDL_PAD_ENET_RXD1__ENET_RX_DATA1	0x100b0
-				MX6QDL_PAD_ENET_TXD0__ENET_TX_DATA0	0x100b0
-				MX6QDL_PAD_ENET_TXD1__ENET_TX_DATA1	0x100b0
-				MX6QDL_PAD_GPIO_16__ENET_REF_CLK	0x400000c0
+				/* MDIO speed up to 5 MHz */
+				MX6QDL_PAD_ENET_MDIO__ENET_MDIO		0x100b1
+				MX6QDL_PAD_ENET_MDC__ENET_MDC		0x000b1
+				/* RMII 50 MHz */
+				MX6QDL_PAD_ENET_CRS_DV__ENET_RX_EN	0x100f5
+				MX6QDL_PAD_ENET_TX_EN__ENET_TX_EN	0x100f5
+				MX6QDL_PAD_ENET_RXD0__ENET_RX_DATA0	0x100c0
+				MX6QDL_PAD_ENET_RXD1__ENET_RX_DATA1	0x100c0
+				MX6QDL_PAD_ENET_TXD0__ENET_TX_DATA0	0x100f5
+				MX6QDL_PAD_ENET_TXD1__ENET_TX_DATA1	0x100f5
+				MX6QDL_PAD_GPIO_16__ENET_REF_CLK	0x40010040
+				MX6QDL_PAD_GPIO_5__GPIO1_IO05		0x40000058
+				/* GPIO for "link active" */
+				MX6QDL_PAD_ENET_RX_ER__GPIO1_IO24	0x40000058
 			>;
 		};
 
